core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/LDAPConnectionContext.java

```diff
@@ -211,14 +211,20 @@ public LDAPConnectionContext(RealmConfiguration realmConfig) throws UserStoreExc
 
         // Set socket factory for not verifying SSL certificate.
         boolean isSSLCertificateValidationEnabled = true;
-        String valueForSSLCertificateValidationEnabled = realmConfig.getUserStoreProperty(UserStoreConfigConstants.
-                SSLCertificateValidationEnabled);
-        if (StringUtils.isNotEmpty(valueForSSLCertificateValidationEnabled)) {
-            isSSLCertificateValidationEnabled = Boolean.parseBoolean(valueForSSLCertificateValidationEnabled);
+        boolean isSecuredLDAP = false;
+        if (connectionURL != null) {
+            isSecuredLDAP = connectionURL.startsWith("ldaps:");
         }
+        if (isSecuredLDAP) {
+            String valueForSSLCertificateValidationEnabled = realmConfig.getUserStoreProperty(UserStoreConfigConstants.
```

Comment:
If the `SSLCertificateValidationEnabled` config is properly set (false for ldap connection), this should have been working fine. Do you know why that is not working?

Reply:
The reason is that the NonVerifyingSSLSocketFactory is still a ssl socket factory. Setting this property just when this config is unticked for ldap as well causes this issue.
This config should matter only if this is a secured connection